code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -276,7 +276,7 @@
 				"You hear a welding torch on metal."
 			)
 			playsound(loc, 'sound/items/Welder2.ogg', 50, 1)
-			if (!do_after(user, 2 SECONDS, act_target = src, extra_checks = CALLBACK(src, .proc/is_closed)))
+			if (!do_after(user, 2/W.toolspeed, act_target = src, extra_checks = CALLBACK(src, .proc/is_closed)))

```

Comment:
This change will make it happen after 2 ticks / toolspeed instead of 2 seconds / toolspeed